client/src/views/client-area/payment-methods/CreditCardModal.vue

```diff
@@ -33,22 +33,26 @@
 <script setup lang="ts">
 import { MsModal, MsModalResult, MsStripeCardForm, PaymentMethodResult, I18n, Translatable } from 'megashark-lib';
 import { IonToggle, modalController } from '@ionic/vue';
-import { ref } from 'vue';
+import { ref, useTemplateRef } from 'vue';
 import { BmsAccessInstance } from '@/services/bms';
 import { Information, InformationLevel, InformationManager, PresentationMode } from '@/services/informationManager';
 
 const props = defineProps<{
   informationManager: InformationManager;
 }>();
 
-const cardForm = ref();
+const cardFormRef = useTemplateRef<InstanceType<typeof MsStripeCardForm>>('cardForm');
 const setDefault = ref(false);
 const error = ref<Translatable | undefined>(undefined);
 const querying = ref(false);
 
 async function confirm(): Promise<boolean> {
   querying.value = true;
-  const result: PaymentMethodResult = await cardForm.value.submit();
+  const result: PaymentMethodResult | undefined = await cardFormRef.value!.submit();
```

Comment:
A `try/finally` would be good here to set querying to false. A ton of conditions, it's very easy to forget.